editor.planx.uk/src/pages/FlowEditor/lib/store/user.ts

```diff
@@ -0,0 +1,43 @@
+import { Team, User } from "types";
+import type { StateCreator } from "zustand";
+
+export interface UserStore {
+  name: string;
+  email: string;
+  isPlatformAdmin: boolean;
+  roles: { teamSlug: Team["slug"]; role: "teamEditor" | "teamViewer" }[] | [];
+  isViewOnly?: boolean;
+
+  setUser: (user: User) => void;
+  getUser: () => User;
+}
+
+export const userStore: StateCreator<UserStore, [], [], UserStore> = (
+  set,
+  get,
+) => ({
+  name: "",
+  email: "",
+  isPlatformAdmin: false,
+  roles: [],
+
+  setUser: (user) =>
+    set({
+      name: user.name,
+      email: user.email,
+      isPlatformAdmin: user.isPlatformAdmin,
+      roles: user.roles,
+    }),
+
+  getUser: () => ({
+    name: get().name,
+    email: get().email,
+    roles: get().roles,
+    isPlatformAdmin: get().isPlatformAdmin,
+    isViewOnly:
+      !get().isPlatformAdmin &&
+      !get()
+        .roles.map((r) => r.role)
+        .includes("teamEditor"),
```

Comment:
i'm not actually sure we want `isViewOnly` on the actual User type, rather it's probably better as it's own method within the user store here, but again this is just quick first go !